src/Hamiltonians/angular_momentum.jl

```diff
@@ -0,0 +1,90 @@
+"""
+    AMzProjectionHO(S; z_dim = 3) <: AbstractHamiltonian
```

Comment:
Can we call it `AngularMomentumProjectionHO`?

Reply:
or how about `LzProjectionHO`
```suggestion
    LzProjectionHO(S; z_dim = 3) <: AbstractHamiltonian
```